src/FixedDataTableBufferedRows.js

```diff
@@ -78,27 +81,32 @@ class FixedDataTableBufferedRows extends React.Component {
     var rowClassNameGetter = props.rowClassNameGetter || emptyFunction;
     var rowsToRender = this.props.rowsToRender || [];
 
-    this._staticRowArray.length = rowsToRender.length;
+    if (props.isScrolling) {
+      // we are scrolling, so don't display any rows which lie outside the viewport
+      this._staticRowArray.forEach((row, i) => {
+        const rowOutsideViewport = !this.isRowInsideViewport(row.props.index);
+        if (rowOutsideViewport) {
+          this._staticRowArray[i] = React.cloneElement(this._staticRowArray[i], {
+            visible: false,
+          });
+        }
+      });
+    } else {
+      this._staticRowArray.length = rowsToRender.length;
+    }
+
     var baseOffsetTop = props.offsetTop - props.scrollTop;
 
-    for (let i = 0; i < rowsToRender.length; ++i) {
+    for (let i = 0; i < rowsToRender.length; i++) {
       const rowIndex = rowsToRender[i];
+
+      // if the row doesn't exist in the buffer, assign a fake row to it.
+      // this is so that we can get rid of unnecessary row mounts/unmounts
       if (rowIndex === undefined) {
-        this._staticRowArray[i] = (
```

Comment:
moved these kind of row creation logic into this.getFakeRow (see [comment](https://github.com/schrodinger/fixed-data-table-2/pull/415#discussion_r270716208))